In [2]:
import seaborn as sns

In [3]:
df = sns.load_dataset('titanic')

In [61]:
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

# 타이타닉 데이터 컬럼 설명
- survived: 생존 여부 / 0이면 사망, 1이면 생존
- pclass: 객실 등급 / 1이면 1등급, 2이면 2등급, 3이면 3등급
- sex: 성별 / male이면 남자, female이면 여자
- age: 연령
- sibsp: 타이타닉호에 탑승한 형제 및 배우자의 수
- parch: 타이타닉호에 탑승한 부모 및 자녀의 수
- fare: 승객 요금
- embarked: 탑승지 이름 앞글자 / C(Cherbourg), Q(Queenstown), S(Southampton)
- class: 객실 등급 / First면 1등급, Second면 2등급, Third면 3등급
- who:남자, 여자, 아이 / man, woman, child
- adult_male: 성인 남자인지 여부 / True면 성인 남자, False면 그외
- deck: 선실 번호 첫 알파벳 / A,B,C,D,E,F,G
- embark_town: 탑승지 이름 / Cherbourg, Queenstown, Southampton
- alive: 생존 여부 / no면 사망, yes면 생존
- alone: 혼자 탑승했는지 여부 / True면 혼자 탑승, False면 가족과 함께 탑승
-------------------------------------------------------------------------------------------------------------------------

# 질문 10가지

1. 생존자/미생존자 각 인원은?
2. 생존자 중 가장 많이 탑승한 곳은?
3. 생존자/미생존자의 평균 연령을 제시하시오
4. 생존자 중 미성년자를 정렬하시오 (오름차순으로)
5. 남녀 수를 생존자별로 제시하시오
6. 생존자들 객실 등급의 각 개수를 구하시오
7. 생존자들의 선실 번호를 정렬하시오 (오름차순으로)
8. 형제 및 배우자 그리고 부모 및 자녀와 함께 탑승한 생존자들의 인원을 각각 제시하시오
9. 20대부터 40대까지의 생존자 중 혼자 오신 분의 인원은?
10. 생존한 미성년자를 'yes'로, 아니면 'no'로 표현하여 제시하시오 (새로운 변수 이름: survived_child)

-------------------------------------------------------------------------------------------------------------------------
1. 생존자/미생존자 각 인원은?

In [62]:
df.groupby('survived').agg(n = ('survived', 'count'))

n
survived     
0         549
1         342

생존자: 342명, 미생존자: 549명

-------------------------------------------------------------------------------------------------------------------------
2. 생존자 중 가장 많이 탑승한 곳은?

In [63]:
df.query("survived == 1").groupby('embark_town').agg(n = ('embark_town', 'count'))

n
embark_town     
Cherbourg     93
Queenstown    30
Southampton  217

Southampton

-------------------------------------------------------------------------------------------------------------------------
3. 생존자/미생존자의 평균 연령을 제시하시오

In [64]:
df.groupby('survived').agg(mean_age = ('age', 'mean'))

mean_age
survived           
0         30.626179
1         28.343690

-------------------------------------------------------------------------------------------------------------------------
4. 생존자 중 미성년자를 정렬하시오 (오름차순으로)

In [65]:
df.query("survived == 1 and age <= 19")[['age']].sort_values('age')

age
803   0.42
755   0.67
644   0.75
469   0.75
831   0.83
..     ...
283  19.00
291  19.00
546  19.00
136  19.00
887  19.00

[79 rows x 1 columns]

-------------------------------------------------------------------------------------------------------------------------
5. 남녀 수를 생존자별로 제시하시오

In [66]:
df.groupby(['survived', 'sex']).agg(n=('survived', 'count'))

n
survived sex        
0        female   81
         male    468
1        female  233
         male    109

-------------------------------------------------------------------------------------------------------------------------
6. 생존자들 객실 등급의 각 개수를 구하시오

In [67]:
df.query("survived == 1").groupby('class').agg(n = ('class', 'count'))

n
class      
First   136
Second   87
Third   119

First: 136개, Second: 87개, Third: 119개

-------------------------------------------------------------------------------------------------------------------------
7. 생존자들의 선실 번호를 정렬하시오 (오름차순으로)

In [68]:
df.query("survived == 1")[['deck']].sort_values('deck')

deck
556    A
445    A
209    A
647    A
23     A
..   ...
866  NaN
869  NaN
874  NaN
875  NaN
880  NaN

[342 rows x 1 columns]

-------------------------------------------------------------------------------------------------------------------------
8. 형제 및 배우자 그리고 부모 및 자녀와 함께 탑승한 생존자들의 인원을 각각 제시하시오

In [69]:
df.query("survived==1 and sibsp >= 1 and parch >= 1").groupby(['sibsp', 'parch']).agg(n_sibsp = ('sibsp', 'count'),
                                                                                     n_parch = ('parch', 'count'))

n_sibsp  n_parch
sibsp parch                  
1     1           34       34
      2           12       12
      3            1        1
      5            1        1
2     1            6        6
      2            2        2
      3            1        1
3     2            2        2
4     2            3        3

-------------------------------------------------------------------------------------------------------------------------
9. 20대부터 40대까지의 생존자 중 혼자 오신 분의 인원은?

In [72]:
df.query("age >= 20 and age < 50 and survived == 1").groupby('alone').agg(n = ('alone', 'count'))

n
alone     
False   84
True   100

100명

-------------------------------------------------------------------------------------------------------------------------
10. 생존한 미성년자를 'yes'로, 아니면 'no'로 표현하여 제시하시오 (새로운 변수 이름: survived_child)

In [71]:
import numpy as np
df.assign(survived_child = np.where((df['survived'] == 1) & (df['age']<=19), 'yes', 'no'))

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone survived_child  
0      man        True  NaN  Southampton    no  False             no  
1    woman       False    C    Cherbourg   yes  False             no  
2    woman       False  NaN  Southampton   yes   True             no  
3    woman       False    C  Southampton   yes  False             no  
4      man        True  NaN  Southampton    no   True             no  
..     ...         ...  ...          ...   ...    ...            ...  
886    man        True  NaN  Southampton    no   True             no  
887  woman       False    B  Southampton   yes   True            yes  
888  woman       False  NaN  Southampton    no  False             no  
889    man        True    C    Cherbourg   yes   True             no  
890    man        True  NaN   Queenstown    no   True             no  

[891 rows x 16 columns]

-------------------------------------------------------------------------------------------------------------------------